### PR TITLE
4 add new device

### PR DESCRIPTION
**Pull request by ArtemKlots-in6k**

### CODE REVIEW

File: api/src/main/java/com/in6k/vault13/controller/DeviceController.java [hunk 1]
```diff
@@ -49,4 +42,17 @@ public ResponseEntity createDevice(@RequestBody Device device) throws Exception
         return ResponseEntity.ok(deviceTypeList);
     }
 
+    @RequestMapping(value = "/devices/empty", method = RequestMethod.GET)
+    public ResponseEntity<Device> createEmptyDevice() throws ArangoException {
+        Device emptyDevice = deviceService.createDefaultDevice();
```
Comment (alexanderko) [inline api/src/main/java/com/in6k/vault13/controller/DeviceController.java]: `default` is not appropriate name here. Please use `empty` instead.

File: api/src/main/java/com/in6k/vault13/dao/DevicesDao.java [hunk 2]
```diff
@@ -2,14 +2,21 @@
...
     List<Device> getAll() throws Exception;
 
-    void add(Device device);
+    Device create(Device device);
 
     Device getBySerialNumber(String serialNumber) throws ArangoException;
+
+    Device createDefaultDevice();
+
+    void deleteByID(Device device);
```
Comment (alexanderko) [inline api/src/main/java/com/in6k/vault13/dao/DevicesDao.java]: Method says it delates by id but it accepts device object. Rename method to just delete.

File: api/src/main/java/com/in6k/vault13/dao/ModelDao.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+import java.util.List;
+
+public interface ModelDao {
+
+    List<Model> getAll();
+
+    Model create(Model model, Vendor vendor);
+
+    void assignVendorToDevice(Model model, Vendor vendor);
```
Comment (alexanderko) [inline api/src/main/java/com/in6k/vault13/dao/ModelDao.java]: refactor method name

File: api/src/main/java/com/in6k/vault13/dao/impl/DeviceTypeDaoImpl.java [hunk 4]
```diff
@@ -1,39 +1,39 @@
...
+        return DaoUtils.getResult(databaseConnector.getArangoDriver(), query, clazz);
     }
 
     @Override
     public void add(DeviceType deviceType) {
         BaseDocument type = new BaseDocument();
         type.addAttribute("title", deviceType.getTitle());
         try {
-            databaseConnector.getArangoDriver().createDocument("typesOfDevices", type);
+            databaseConnector.getArangoDriver().createDocument(ArangoCollections.TYPES_OF_DEVICES.getName(), type);
```
Comment (alexanderko) [inline api/src/main/java/com/in6k/vault13/dao/impl/DeviceTypeDaoImpl.java]: rename collection to `device_types`

File: db/init-data.sh [hunk 5]
```diff
@@ -24,3 +31,37 @@ arangoimp \
...
+     --create-collection-type edge
+
+arangoimp \
+--server.endpoint unix:///tmp/arangodb-tmp.sock \
+--server.username root \
+ --server.password "$ARANGO_ROOT_PASSWORD" \
+  --file "$INIT_DATA_DIR/produced-by-edges.json" \
+   --type json --collection "produced_by_edges" \
+    --create-collection true \
+     --create-collection-type edge
```
Comment (alexanderko) [inline db/init-data.sh]: fix indents

File: client/src/app/devices/mock-device.ts [hunk 6]
```diff
@@ -2,11 +2,11 @@ import {Device} from './device';
...
-    {id: "5", type: TYPES[4], producer: "q", model: "model", serialNumber: "r", notes: "t"},
-    {id: "6", type: TYPES[5], producer: "q", model: "model", serialNumber: "r", notes: "t"},
-    {id: "7", type: TYPES[6], producer: "q", model: "model", serialNumber: "r", notes: "t"}
+/*    {id: "1", type: TYPES[0], vendor: {id:""}, model: "model", serialNumber: "r", notes: "t"},
+    {id: "2", type: TYPES[1], vendor: "q", model: "model", serialNumber: "r", notes: "t"},
+    {id: "3", type: TYPES[2], vendor: "q", model: "model", serialNumber: "r", notes: "t"},
+    {id: "4", type: TYPES[3], vendor: "q", model: "model", serialNumber: "r", notes: "t"},
+    {id: "5", type: TYPES[4], vendor: "q", model: "model", serialNumber: "r", notes: "t"},
+    {id: "6", type: TYPES[5], vendor: "q", model: "model", serialNumber: "r", notes: "t"},
+    {id: "7", type: TYPES[6], vendor: "q", model: "model", serialNumber: "r", notes: "t"}*/
```
Comment (alexanderko) [inline client/src/app/devices/mock-device.ts]: remove useless service

File: client/src/app/devices/types/mock-types.ts [hunk 7]
```diff
@@ -1,14 +1,14 @@
...
+    {id: "11", name: 'Computer'},
+    {id: "12", name: 'Monitor'},
+    {id: "13", name: 'Mouse'},
+    {id: "14", name: 'Keyboard'},
+    {id: "15", name: 'Magneta'},
+    {id: "16", name: 'RubberMan'},
+    {id: "17", name: 'Dynama'},
+    {id: "18", name: 'Dr IQ'},
+    {id: "19", name: 'Magma'},
+    {id: "20", name: 'Tornado'}
```
Comment (alexanderko) [inline client/src/app/devices/types/mock-types.ts]: remove this mock

File: api/src/main/java/com/in6k/vault13/controller/DeviceController.java (hunk 1 above)
Comment (ArtemKlots-in6k) [inline api/src/main/java/com/in6k/vault13/controller/DeviceController.java]: Done

File: api/src/main/java/com/in6k/vault13/dao/DevicesDao.java (hunk 2 above)
Comment (ArtemKlots-in6k) [inline api/src/main/java/com/in6k/vault13/dao/DevicesDao.java]: Done

File: api/src/main/java/com/in6k/vault13/dao/ModelDao.java (hunk 3 above)
Comment (ArtemKlots-in6k) [inline api/src/main/java/com/in6k/vault13/dao/ModelDao.java]: Done, если переименовать нужно было только assignVendorToDevice. Переименовал в assignVendorToDevice и переместил в VendorDao (логически-правильное расположение)

File: api/src/main/java/com/in6k/vault13/dao/impl/DeviceTypeDaoImpl.java (hunk 4 above)
Comment (ArtemKlots-in6k) [inline api/src/main/java/com/in6k/vault13/dao/impl/DeviceTypeDaoImpl.java]: Done

**Pull request merged**